src/controllers/UsersController.php

```diff
@@ -267,4 +267,26 @@ public function updateUser(Request $request, $db)
         }
         throw new Exception("Could not update user", 400);
     }
+
+    public function passwordReset(Request $request, $db)
+    {
+        $token = filter_var($request->getParameter("token"), FILTER_SANITIZE_STRING);
+        if(empty($token)) {
+            throw new Exception("Reset token must be supplied", 400);
+        }
+
+        $password = $request->getParameter("password");
+        if(empty($password)) {
+            throw new Exception("New password must be supplied", 400);
+        }
+
```

Comment:
Need to call `checkPasswordValidity()` here as per UsersController::postAction().